### ISSUE TITLE
Nepočítat DPH pokud jsem plátce daně

### ISSUE THREAD

**Issue by vyskoczilova** [vyskoczilova/kybernaut-ic-dic#11]
##Patrik O.##

Dobrý den,

v prvé řadě chci pochválit Váš plugin Kybernaut IČ DIČ a dále a chci se zeptat, zda by do pluginu nešel nějakým způsobem přidat filter či nějak zakomponovat, aby se v případě, že je zákazník (odběratel) plátce DPH (tzn. validní DIČ), vymazala sazba daně (pokud je tedy dodavatel také plátce DPH - tato možnost by mohla být nejlépe asi zvolena přímo v administraci - JSEM PLÁTCE DPH.

A poté, pokud by byla splněna tato podmínka a zároveň podmínka,že odběratel je plátce DPH, tak by se sazba daně znulovala, jednoduše přes 

`$woocommerce->customer->set_is_vat_exempt( true ) ;`

Děkuji mnohokrát

## KV ##

To je dobrý nápad. Zvládnete připsat pull request a poslat mi ho na githubu?

## Patrik O. ##

Děkuji mnohokrát za odpověď.

Díval jsem se trochu víc podrobně na kód Vašeho pluginu. Nejsem tedy žádný profesionál, co se programování týká, ale zdá se, že by se 

$woocommerce->customer->set_is_vat_exempt( true ) ; 

muselo přidat do podmínky (soubor filters-actions.php), pouze v případě, že  zákazník (odběratel), je plátce DPH, ale je mimo ČR, či prostě jinou zemi v EU než je adresa obchodu (dodavatele)).

Dovolte mi trochu konkrétněji nastínit situace, protože má předchozí formulace byla jen stručná a mohla působit nepřesně.

**Konkrétně nastávají tyto situace:**

a) obchod (dodavatel) není plátce DPH --> neměly by být vůbec nastaveny sazby daně, takže tady se nic neřeší

b) obchod (dodavatel) je plátce DPH -->  odběratel je plátce DPH (má platné VAT či DIČ chcete-li) && jeho adresa je v EU && země odběratele se liší od země v adrese obchodu, poté platí
`$woocommerce->customer->set_is_vat_exempt( true ) ;   //daň nebude započítána`

c) obchod (dodavatel) je plátce DPH -->  odběratel je plátce DPH (má platné VAT či DIČ chcete-li) && jeho adresa je v EU &&  země se NEliší od země v adrese obchodu, poté platí
`$woocommerce->customer->set_is_vat_exempt( false ) ;    //daň bude započítána`

d) obchod (dodavatel) je plátce DPH -->  odběratel  nemá platné VAT či DIČ chcete-li || jeho adresa není v EU 
`$woocommerce->customer->set_is_vat_exempt( false ) ;    //daň bude započítána`

Nevím, jestli je jednodušší nadefinovat proměnou plátce DPH, která by nabírala hodnoty true pokud by si to uživatel zatrhnul v nastavení Vašeho pluginu či zda to prostě rozlišit pouze tak, že pokud jsou daňové sazby povoleny (tak logicky je pak obchod plátce DPH), na to by snad měla fungovat funkce 
`wc_tax_enabled();`
https://docs.woocommerce.com/wc-apidocs/function-wc_tax_enabled.html

S Gitem právě začínám, tak uvidím, zda toho budu schopen :)

P.O.

**Comment by Musilda** [comment]
Ahoj, jak to s tímto vypadá? Zatím to obcházím kombinací tvého pluginu a ještě jednoho, kdy skrývám pole pro DIČ.

**Comment by Musilda** [comment]
Nebo by mi stačilo možnost pole pro DIČ úplně vypnout.

**Comment by vyskoczilova** [comment]
Ahoj, díky vypadá to tak, že mám rozepsaný refaktor pluginu a nedostatek času, protože dodělávám nějaké onlinové věci pro stálé klienty.

Jestli ti stačí vypnutí, tak mi přidej pull request s filtrem a já to pošlu do pluginu rovnou.

**Comment by Musilda** [comment]
Ok, až se utrhnu od klientů, tak to udělám. Stejně to jeden z nich chce.

**Comment by vyskoczilova** [comment]
Ok, tak jo. Díky moc.

**Comment by morvy** [comment]
No, ako som uz aj pri inom issue pisal, ja toto uz mam relativne stabilne zapracovane na webe, len sa neviem dopracovat k tomu aby som si nasiel cas a hodil pull :(

**Comment by vyskoczilova** [comment]
Připraveno pro release 1.7.0 díky @morvy
